File: backend/app/schemas/rsvp.py
from pydantic import BaseModel, Field


class RsvpIn(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    attending: bool
    guests_count: int = Field(ge=1, le=20)
    message: str | None = Field(default=None, max_length=2000)


class RsvpOut(BaseModel):
    id: int
    name: str
    attending: bool
    guests_count: int
    message: str | None = None

    class Config:
        from_attributes = True

File: frontend/src/stores/rsvp.ts
import { defineStore } from 'pinia'
import { postJson, type ApiError } from '../services/api'

export type RsvpIn = {
  name: string
  attending: boolean
  guests_count: number
  message?: string | null
}

export type RsvpOut = {
  id: number
  name: string
  attending: boolean
  guests_count: number
  message?: string | null
}

export const useRsvpStore = defineStore('rsvp', {
  state: () => ({
    submitting: false,
    error: null as string | null,
  }),
  actions: {
    async submit(payload: RsvpIn) {
      this.submitting = true
      this.error = null
      try {
        return await postJson<RsvpIn, RsvpOut>('/api/rsvp', payload)
      } catch (e) {
        const err = e as ApiError
        this.error = err.message ?? 'Impossible d’envoyer votre réponse.'
        throw e
      } finally {
        this.submitting = false
      }
    },
  },
})

